### DISCUSSION TITLE
RichText not following correct colors

### DISCUSSION THREAD

**Discussion by DanMossa** [rydmike/flex_color_scheme#43] [Q&A] [answered]
I have a `RichText` widget

```
  RichText(
    text: TextSpan(
        text: widget.profileStore.milesBetween.toString(),
        style: const TextStyle(
          fontWeight: FontWeight.bold,
          fontSize: 18.0,
          //color: Color(0xff724c0b),
        ),
        children: const [
          TextSpan(
              text: "mi",
              style: TextStyle(
                  color: Colors.grey, fontSize: 12.0))
        ]),
  ),
```

The color is defaulting to white and not the normal color that occurs inside of `Text` widgets.
![image](https://user-images.githubusercontent.com/10294777/148300488-818c4978-f915-4ed6-a107-bdae1b74c591.png)

**Comment by rydmike** [comment]
Hi @DanMossa thanks for opening this issue.
Can you share a bit more information about your FlexColorScheme setup? Like did you opt-in on the opinionated sub themes and there the M3 styled TextTheme? Was this in dark or light theme mode?

Regardless of if you can share any more info. I will setup a test case and try to replicate the issue as well and get back to you so we can figure out together if it is a bug or config/setup related issue.

**Comment by rydmike** [comment, marked as answer]
Hi @DanMossa I reviewed your case. In your example you are using the `RichText` widget. It is a lower level widget that does not use any text theme, it must always be explicitly styled as explained here in Flutter docs: https://api.flutter.dev/flutter/widgets/RichText-class.html

If you want to use rich text that by default depends on the surrounding theme and its text theme, you should use `Text.rich` instead.

Here is an example based on your example without setting any theme color on the first number text:

```dart
class DemoRichText extends StatelessWidget {
  const DemoRichText({Key? key}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return RichText(
      text: const TextSpan(
          text: '200.5',
          style: TextStyle(
            fontWeight: FontWeight.bold,
            fontSize: 18.0,
          ),
          children: <InlineSpan>[
            TextSpan(
                text: ' miles',
                style: TextStyle(color: Colors.grey, fontSize: 12.0))
          ]),
    );
  }
}


class DemoTextRich extends StatelessWidget {
  const DemoTextRich({Key? key}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return const Text.rich(
      TextSpan(
        text: '200.5',
        style: TextStyle(
          fontWeight: FontWeight.bold,
          fontSize: 18.0,
        ),
        children: <InlineSpan>[
          TextSpan(
              text: ' miles',
              style: TextStyle(color: Colors.grey, fontSize: 12.0))
        ],
      ),
    );
  }
}
```

The first one `DemoRichText` always produces white numbered text by default, in both light and dark theme mode, this happens when using only Flutter standard themes without FlexColorScheme as well.

The 2nd widget `DemoTextRich` produces a surrounding themed text for the numbers, so it will be black in light theme and white in dark theme. In FlexColorScheme even when using M3 inspired style, its styles will also be correctly applied to it then.

Here the above two widgets are inserted into a UI for demonstration. First in light mode, where the number on the `DemoRichText` cannot be seen since it is white by default. However `DemoTextRich` is dark by default and can be seen:
![image](https://user-images.githubusercontent.com/39990307/148466237-7415413e-6cf6-47a0-94df-7f352b3e8333.png)

Here are the same widgets in dark mode, now the `DemoRichText` can be seen, since it remained white as before, but so can also `DemoTextRich` since it automatically adjusted and followed the surrounding ambient text theme.

![image](https://user-images.githubusercontent.com/39990307/148466471-e0996b77-9cc7-42db-89d9-4d5911845987.png)


Hope this helps, the behavior you are seeing is actually expected and normal behavior with the widget you used.
Feel free to close the issue if the above answer is satisfactory and solved your issue.

Kind regards, Mike

**Comment by DanMossa** [comment]
@rydmike 
Sorry for the late response, I haven't been near my computer.

You are an absolute wizard. You're 100% right and using `Text.rich` fixed the issue I was having. Thank you!
